js_modules/dagster-ui/packages/ui-core/src/asset-selection/__tests__/AntlrAssetSelection.test.ts

```diff
@@ -52,6 +52,21 @@ function assertQueryResult(query: string, expectedNames: string[]) {
 }
 
 describe('parseAssetSelectionQuery', () => {
+  describe('invalid queries', () => {
+    it('should throw on invalid queries', () => {
+      expect(() => parseAssetSelectionQuery(TEST_GRAPH, 'A')).toThrow();
+      expect(() => parseAssetSelectionQuery(TEST_GRAPH, 'key:A key:B')).toThrow();
+      expect(() => parseAssetSelectionQuery(TEST_GRAPH, 'not')).toThrow();
+      expect(() => parseAssetSelectionQuery(TEST_GRAPH, 'and')).toThrow();
+      expect(() => parseAssetSelectionQuery(TEST_GRAPH, 'key:A and')).toThrow();
+      expect(() => parseAssetSelectionQuery(TEST_GRAPH, 'sinks')).toThrow();
+      expect(() => parseAssetSelectionQuery(TEST_GRAPH, 'notafunction()')).toThrow();
+      expect(() => parseAssetSelectionQuery(TEST_GRAPH, 'tag:foo=')).toThrow();
+      expect(() => parseAssetSelectionQuery(TEST_GRAPH, 'owner')).toThrow();
+      expect(() => parseAssetSelectionQuery(TEST_GRAPH, 'owner:owner@owner.com')).toThrow();
+    });
```

Comment:
does this function throw even without your change above?

Reply:
no, the default error listener only logs the error message